server/src/main/java/org/elasticsearch/index/mapper/DateFieldMapper.java

```diff
@@ -692,6 +692,10 @@ public Relation isFieldWithinQuery(
             DateMathParser dateParser,
             QueryRewriteContext context
         ) throws IOException {
+            if (reader == null) {
+                // NOTE: in case we rewrite a query before we have a reader available
+                return Relation.INTERSECTS;
```

Comment:
is this uncovering an existing bug that may be triggered by the percolator, or only caused by doing a rewrite before acquiring a searcher?

Reply:
No bug, just doing a rewrite before we have a searcher available.